src/mdns/mdns.hpp

```diff
@@ -77,6 +60,50 @@ typedef void (*StateHandler)(void *aContext, State aState);
 class Publisher
 {
 public:
+    /**
+     * This structure represents a name/value pair of the TXT record.
+     *
+     */
+    struct TxtEntry
+    {
+        const char *   mName;        ///< The name of the TXT entry.
```

Comment:
Storing the pointer assumes the lifetime of the memory passed to the `TxtEntry`. Is it more favorable to use `std::unique_ptr<char[]>`?

Reply:
I think it applies to other arguments: `const char *aName`, `const char *aType`. `std::unique_ptr` doesn't solve the problem unless we make a copy of the string and in this case we should use `std::string`. Thoughts?